nca/Parsers/IngressPolicyYamlParser.py

```diff
@@ -155,7 +155,7 @@ def segregate_longest_paths_and_make_dfa(parsed_paths):
                 path_regex = path_string
             else:
                 if path_string:
-                    path_regex = path_string + '|' + path_string + '/' + allowed_chars + '*'
+                    path_regex = path_string + '/' + allowed_chars + '*'
```

Comment:
I believe the `/` should not be mandatory. The spec says `/foo` should match `/foo` even if `pathType` is `Prefix`.
Maybe the regex should be something like `path_string + '(/' + allowed_chars + '*)?'`

Reply:
Right, changed to this pattern..
Is it readable enough? Could it help to convert to an expression such as `[Prefix]/foo' or `[Exact]/foo` ?